src/PagerView.tsx

```diff
@@ -1,155 +1,207 @@
+    this.props.onPageScroll?.(event);
+    if (this.props.keyboardDismissMode === 'on-drag') {
+      Keyboard.dismiss();
+    }
+  };
+
+  private onPageScrollStateChanged = (
+    event: PageScrollStateChangedNativeEvent
+  ) => {
+    this.props.onPageScrollStateChanged?.(event);
+    this.isScrolling = event.nativeEvent.pageScrollState === 'dragging';
   };
 
-  private _onMoveShouldSetResponderCapture = () => {
-    return this.isScrolling;
+  private onPageSelected = (event: PagerViewOnPageSelectedEvent) => {
+    const currentPage = event.nativeEvent.position;
+    this.setState((prevState) =>
+      this.computeRenderWindow({
+        buffer: this.props.buffer,
+        currentPage,
+        maxRenderWindow: this.props.maxRenderWindow,
+        offset: prevState.offset,
+        windowLength: prevState.windowLength,
+      })
+    );
+    this.props.onPageSelected?.(event);
   };
 
+  private renderChildren(offset: number, windowLength: number) {
+    const keys: string[] = [];
+    return {
+      children: this.props.data
+        .slice(offset, offset + windowLength)
+        .map((item, index) => {
+          const key = this.props.keyExtractor(item, offset + index);
+          keys.push(key);
+          return (
+            <View collapsable={false} key={key} style={styles.pageContainer}>
+              {this.props.renderItem({ item, index: offset + index })}
+            </View>
+          );
+        }),
+      keys,
+    };
+  }
+
   render() {
+    const { offset, windowLength } = this.state;
+    const { children, keys } = this.renderChildren(offset, windowLength);
+
     return (
-      <PagerViewViewManager
-        {...this.props}
-        ref={this.PagerView as any /** TODO: Fix ref type */}
+      <PagerViewNative
+        childrenKeys={keys}
+        count={this.props.data.length}
+        offscreenPageLimit={this.props.offscreenPageLimit}
+        offset={offset}
+        onMoveShouldSetResponderCapture={this.onMoveShouldSetResponderCapture}
+        onPageScroll={this.onPageScroll}
+        onPageScrollStateChanged={this.onPageScrollStateChanged}
+        onPageSelected={this.onPageSelected}
+        orientation={this.props.orientation}
+        overdrag={this.props.overdrag}
+        pageMargin={this.props.pageMargin}
+        scrollEnabled={this.props.scrollEnabled}
+        showPageIndicator={this.props.showPageIndicator}
         style={this.props.style}
-        onPageScroll={this._onPageScroll}
-        onPageScrollStateChanged={this._onPageScrollStateChanged}
-        onPageSelected={this._onPageSelected}
-        onMoveShouldSetResponderCapture={this._onMoveShouldSetResponderCapture}
-        children={childrenWithOverriddenStyle(this.props.children)}
-      />
+        transitionStyle={this.props.transitionStyle}
+      >
+        {children}
+      </PagerViewNative>
     );
   }
 }
+
+const styles = StyleSheet.create({
+  pageContainer: { height: '100%', position: 'absolute', width: '100%' },
```

Comment:
We don't need this style since lib has `childrenWithOverriddenStyle` method

Reply:
My thought was this gives more expected behavior to library users.

From the readme:
> **Attention:** Note that you can only use `View` components as children of `PagerView` (cf. folder */example*)
. For Android if `View` has own children, set prop `collapsable` to false <https://reactnative.dev/img/view#collapsable>, otherwise react-native might remove those children views and  and it's children will be rendered as separate pages

Avoiding using `childrenWithOverriddenStyle()` removes all of those limitations.